spring/src/main/java/io/kubernetes/client/spring/extended/manifests/KubernetesFromYamlProcessor.java

```diff
@@ -0,0 +1,111 @@
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.beans.factory.config.InstantiationAwareBeanPostProcessor;
+import org.springframework.util.ReflectionUtils;
+
+public class KubernetesFromYamlProcessor
+    implements InstantiationAwareBeanPostProcessor, BeanPostProcessor, BeanFactoryAware {
+
+  private static final Logger log = LoggerFactory.getLogger(KubernetesFromYamlProcessor.class);
+
+  private ListableBeanFactory beanFactory;
+
+  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+
+    FromYaml fromYamlAnnotation;
+    try {
+      fromYamlAnnotation = beanFactory.findAnnotationOnBean(beanName, FromYaml.class);
+      if (fromYamlAnnotation != null) {
+        Object loadedObj = loadFromYaml(fromYamlAnnotation.filePath());
+        if (!bean.getClass().isAssignableFrom(loadedObj.getClass())) {
+          log.warn(
+              "Incompatible types for @FromYaml, {} cannot apply to {}",
+              fromYamlAnnotation.filePath(),
+              bean.getClass());
+          return bean;
+        }
+        return loadedObj;
+      }
+    } catch (NoSuchBeanDefinitionException e) {
+      return bean;
+    }
+
+    for (Field field : bean.getClass().getDeclaredFields()) {
+      ReflectionUtils.makeAccessible(field);
+      try {
+        if (field.get(bean) != null) {
+          continue; // field already set, skip processing
+        }
+      } catch (IllegalAccessException e) {
+        log.warn("Failed inject resource for @FromYaml annotated field {}", field, e);
+        continue;
+      }
+
+      fromYamlAnnotation = field.getAnnotation(FromYaml.class);
+      if (fromYamlAnnotation == null) {
+        continue; // skip if the field doesn't have the annotation
+      }
+
+      Object loadedObj = loadFromYaml(fromYamlAnnotation.filePath());
+      ReflectionUtils.setField(field, bean, loadedObj);
+    }
+    return bean;
+  }
+
+  @Override
+  public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
+    this.beanFactory = (ListableBeanFactory) beanFactory;
+  }
+
+  private Object loadFromYaml(String targetFilePath) {
+    Path targetPath = Paths.get(targetFilePath);
+    if (!Files.exists(Paths.get(targetFilePath))) { // checks if it exists on the machine
+      // otherwise use load from classpath resources
+      Path classPath = Paths.get(Resources.getResource(targetFilePath).getPath());
+      if (Files.exists(classPath)) { // use classpath it works
+        targetPath = classPath;
+      } else {
+        throw new BeanCreationException(
+            "No such file " + targetFilePath + " either on the machine or classpaths");
+      }
+    }
+
+    try {
+      String yamlContent = new String(Files.readAllBytes(targetPath));
+      Object loadedObj = Yaml.load(yamlContent);
+      return loadedObj;
+    } catch (IOException e) {
```

Comment:
Should we catch a parse error here too?

Reply:
added catching `YAMLException`, is that right?